aea/aea_builder.py

```diff
@@ -0,0 +1,746 @@
+            logger_name = "aea.{}.skills.{}.{}".format(
+                context.agent_name, skill.configuration.author, skill.configuration.name
+            )
+            skill.skill_context.set_agent_context(context)
+            skill.skill_context._logger = logging.getLogger(logger_name)
+
+    def _check_configuration_not_already_added(self, configuration):
+        if (
+            configuration.component_id
+            in self._package_dependency_manager.all_dependencies
+        ):
+            raise ValueError(
+                "Component {} of type {} already added.".format(
+                    configuration.public_id, configuration.component_type
+                )
+            )
+
+    def _check_package_dependencies(self, configuration):
+        self._package_dependency_manager.check_package_dependencies(configuration)
+
+    @staticmethod
+    def _find_component_directory_from_component_id(
+        aea_project_directory: Path, component_id: ComponentId
+    ):
+        """Find a component directory from component id."""
+        # search in vendor first
+        vendor_package_path = (
+            aea_project_directory
+            / "vendor"
+            / component_id.public_id.author
+            / component_id.component_type.to_plural()
+            / component_id.public_id.name
+        )
+        if vendor_package_path.exists() and vendor_package_path.is_dir():
+            return vendor_package_path
+
+        # search in custom packages.
+        custom_package_path = (
+            aea_project_directory
+            / component_id.component_type.to_plural()
+            / component_id.public_id.name
+        )
+        if custom_package_path.exists() and custom_package_path.is_dir():
+            return custom_package_path
+
+        raise ValueError("Package {} not found.".format(component_id))
+
+    @staticmethod
+    def _try_to_load_agent_configuration_file(aea_project_path: Path):
+        """Try to load the agent configuration file.."""
+        try:
+            configuration_file_path = Path(aea_project_path, DEFAULT_AEA_CONFIG_FILE)
+            with configuration_file_path.open(mode="r", encoding="utf-8") as fp:
+                loader = ConfigLoader.from_configuration_type(ConfigurationType.AGENT)
+                agent_configuration = loader.load(fp)
+                logging.config.dictConfig(agent_configuration.logging_config)
+        except FileNotFoundError:
+            raise Exception(
+                "Agent configuration file '{}' not found in the current directory.".format(
+                    DEFAULT_AEA_CONFIG_FILE
+                )
+            )
+        except jsonschema.exceptions.ValidationError:
+            raise Exception(
+                "Agent configuration file '{}' is invalid. Please check the documentation.".format(
+                    DEFAULT_AEA_CONFIG_FILE
+                )
+            )
+
+    @classmethod
+    def from_aea_project(cls, aea_project_path: PathLike):
```

Comment:
this is used in `aea.cli.run` as a replacement of the old implementation.
Basically, what was done in `aea.cli.run`, `aea.registries` and `aea.skill.base.Skill.from_dir` now it is done here and `aea.configurations.components`. Every component class is then in charge of loading what's relevant for it.

Reply:
see `Component.load_from_directory`